tls/s2n_ktls_io.c

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "tls/s2n_ktls.h"
+
+/* Used to override sendmsg and recvmsg for testing. */
+static ssize_t s2n_ktls_default_sendmsg(struct s2n_connection *conn, struct msghdr *msg, uint8_t record_type);
+static ssize_t s2n_ktls_default_recvmsg(struct s2n_connection *conn, struct msghdr *msg, uint8_t *record_type);
+s2n_ktls_sendmsg_fn s2n_sendmsg_fn = s2n_ktls_default_sendmsg;
+s2n_ktls_recvmsg_fn s2n_recvmsg_fn = s2n_ktls_default_recvmsg;
+
+static ssize_t s2n_ktls_default_recvmsg(struct s2n_connection *conn, struct msghdr *msg, uint8_t *record_type)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(msg);
+    POSIX_ENSURE_REF(record_type);
+    /* kTLS doesn't modify the recv_io_context so its possible to retrieve the socket fd */
```

Comment:
I don't understand this comment. A getter function doesn't usually modify anything.

Reply:
s2n_ktls_get_file_descriptor looks at the `recv_io_context` and gets the fd from that. This is basically saying that its ok to do that because while enabling kTLS, we havnt overwritten the `recv_io_context`.

Reply:
I'm so sorry, I still don't get what this comment is trying to say 😢. If I looked at this without the comment I wouldn't think an explanation is needed.

Reply:
I expanded and moved this comment to `s2n_ktls_get_file_descriptor()`